### PR TITLE
Add pre-commit and github actions

### PR DESCRIPTION
**Pull request by tylerjw**
This PR adds pre-commit and github actions.

### CODE REVIEW

**Comment by tylerjw** [comment]
Here is a PR that tests running the github actions: https://github.com/tylerjw/parameter_tools/pull/1

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+  industrial_ci:
+    strategy:
+      matrix:
+        env:
+          - ROS_DISTRO: galactic
```
Comment (vatanaksoytezer) [inline .github/workflows/ci.yaml:18]: Should we also test for foxy? I see foxy is included in the pre-release yaml.

**Review by vatanaksoytezer — APPROVED** [review]
Just one small question below, otherwise looks good to me, thanks @tylerjw ! Also one thing is I don't see ccov uploaded here, do you happen to know why it might be the case?

**Review by JafarAbdi — APPROVED** [review]

**Comment by tylerjw** [comment]
I'll remove foxy from the preresease.  In my next pr I'm using c++17 features so this really only targets rolling and galactic.

**Comment by tylerjw** [comment]
Ccov will only show up after we merge this.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (tylerjw) [reply .github/workflows/ci.yaml:18]: removed foxy from pre-release testing... this repo is only targeting galactic and rolling as it uses c++17

**Pull request merged**